R/plots.R

```diff
@@ -111,49 +145,69 @@ plot_percent_change_by_age <- function(icb_data, horizon = "2042_43") {
 }
 
-plot_contacts_per_population <- function(icb_data) {
-  list(get_national_contacts(), icb_data[["data"]][[1]]) |>
+plot_count_per_population <- function(icb_data, measure) {
+  list(get_all_national_data(measure), icb_data) |>
     rlang::set_names(c("England", icb_data[["icb22nm"]])) |>
+    purrr::map(pluck_data) |>
     dplyr::bind_rows(.id = "type") |>
     dplyr::mutate(dplyr::across("type", forcats::fct_inorder)) |>
     dplyr::filter(dplyr::if_any("fin_year", \(x) x == "2022_23")) |>
     dplyr::rename(fin_year_popn = "proj_popn_by_fy_age") |>
+    prepare_plot_data(measure) |>
+    dplyr::summarise(
+      dplyr::across("fin_year_popn", unique),
+      dplyr::across("projected_count", sum),
+      .by = c("type", "fin_year", "age_int")
+    ) |>
     add_age_groups() |>
     dplyr::summarise(
-      dplyr::across(c("fin_year_popn", "projected_contacts"), sum),
+      dplyr::across(c("fin_year_popn", "projected_count"), sum),
       .by = c("type", "age_group_cat")
     ) |>
     dplyr::mutate(
-      contacts_rate = .data[["projected_contacts"]] / .data[["fin_year_popn"]],
+      rate = .data[["projected_count"]] / .data[["fin_year_popn"]],
       .keep = "unused"
     ) |>
-    ggplot2::ggplot(ggplot2::aes(.data$age_group_cat, .data$contacts_rate)) +
+    ggplot2::ggplot(ggplot2::aes(.data$age_group_cat, .data$rate)) +
     ggplot2::geom_col(
       ggplot2::aes(fill = .data$type),
       position = "dodge",
       width = 0.75
     ) +
-    ggplot2::labs(x = "Age group", y = "Contacts / 1000 population") +
+    ggplot2::labs(x = "Age group", y = "Count / 1000 population") +
     ggplot2::scale_y_continuous(
-      labels = scales::label_number(scale = 1e3, suffix = "k")
+      labels = scales::label_number(scale = 1e3)
     ) +
     StrategyUnitTheme::scale_fill_su(name = NULL) +
     su_chart_theme()
 }
 
 
-plot_percent_change_by_service <- function(icb_data, horizon = "2042_43") {
-  list(get_national_contacts(), icb_data[["data"]][[1]]) |>
+plot_percent_change_by_service <- function(
+  icb_data,
+  measure,
+  horizon = "2042_43"
+) {
+  list(get_all_national_data(measure), icb_data) |>
+    purrr::map(pluck_data) |>
     rlang::set_names(c("England", icb_data[["icb22nm"]])) |>
     dplyr::bind_rows(.id = "type") |>
-    dplyr::mutate(dplyr::across("type", forcats::fct_inorder)) |>
     dplyr::filter(.data$fin_year %in% c("2022_23", horizon)) |>
+    tidyr::unnest("data") |>
+    dplyr::mutate(
+      dplyr::across("type", forcats::fct_inorder),
+      dplyr::across("service", \(x) tidyr::replace_na(x, "Not recorded"))
+    ) |>
     dplyr::summarise(
-      value = sum(.data[["projected_contacts"]]),
-      .by = c("type", "fin_year", "team_type")
+      value = sum(.data[["projected_count"]]),
+      .by = c("type", "fin_year", "service")
     ) |>
+    # Introduces NAs (which we can then use to remove services from the chart
+    # completely if any value (national or ICB, base or horizon) is missing).
```

Comment:
late night brain didn't flag that if you have a baseline number then you'll have a horizon number... you shouldn't have only one of the pair missing! as the latter is just projected from the former. I can edit the comment...